File: 120-triangle/triangle.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        vector<vector<int>> dp;
        dp.push_back(triangle[0]);
        for(int i=1; i<triangle.size(); i++){
            vector<int> temp(triangle[i].size());
            for(int j=0; j<triangle[i].size(); j++){
                int left = (j-1 >= 0) ? dp[i-1][j-1] : 100000;
                int up = (j < triangle[i-1].size()) ? dp[i-1][j] : 100000;

                temp[j] = triangle[i][j] + min(left, up);
            }
            dp.push_back(temp);
        }

        int ans = 100000;
        for(int i=0; i<dp[dp.size()-1].size(); i++){
            ans = min(dp[dp.size()-1][i], ans);
        }

        return ans;
    }
};

File: 120-triangle/triangle.go
func minimumTotal(triangle [][]int) int {
    m := len(triangle)
    for i := 1; i < m; i++ {
        n := len(triangle[i])

        for j := 0; j < n; j++ {
            prev := math.MaxInt
            if j > 0 {
                prev = triangle[i-1][j-1]
            }
            curr := math.MaxInt
            if j < n-1 {
                curr = triangle[i-1][j]
            }

            triangle[i][j] = triangle[i][j] + min(prev, curr)
        }
    }

    min_path_sum := math.MaxInt
    n := len(triangle[m-1])

    for i := 0; i < n; i++ {
        min_path_sum = min(min_path_sum, triangle[m-1][i])
    }

    return min_path_sum
}